index.js

```diff
@@ -160,12 +156,107 @@ const guessModeHandlers = Alexa.CreateStateHandler(states.GUESSMODE, {
 
 // These handlers are not bound to a state
 const guessAttemptHandlers = {
-  PickACard(cardChoices) {
+  DescribeRecentState(messageSoFarArg, state) {
+    console.log('Player ship:');
+    console.log(JSON.stringify(state.game.ships.playerShip));
+    console.log('AI ship:');
+    console.log(JSON.stringify(state.game.ships.aiShip));
+    console.log('Planet:');
+    console.log(JSON.stringify(state.game.planet));
+    const newChoices = getChoices(state);
+    let messageSoFar = messageSoFarArg;
+    if (state.game.defenseCardChoices) {
+      const playerOffenseChoice = _.values(state.game.offenseCardChoices.playerCards)[0];
+      if (playerOffenseChoice) {
+        messageSoFar += `You have deployed ${playerOffenseChoice.name}. `;
+      }
+      const aiOffenseChoice = _.values(state.game.offenseCardChoices.aiCards)[0];
+      if (aiOffenseChoice) {
+        messageSoFar += `The opponent has deployed ${aiOffenseChoice.name}. `;
+      }
+      if (Object.keys(state.game.defenseCardChoices.playerCards).length > 0) {
+        messageSoFar += 'What defensive tactics would you like to deploy? ';
+      }
+    } else {
+      const playerDefenseChoice = _.values(state.game.prevDefenseCardChoices.playerCards)[0];
+      if (playerDefenseChoice) {
+        messageSoFar += `You have deployed ${playerDefenseChoice.name} for defense. `;
+      }
+      const aiDefenseChoice = _.values(state.game.prevDefenseCardChoices.aiCards)[0];
+      if (aiDefenseChoice) {
+        messageSoFar += `The opponent has defended with ${aiDefenseChoice.name}. `;
+      }
+      if (state.game.offenseCardChoices) {
+        messageSoFar += 'Time for the next offensive. ';
+      }
+    }
+    if (newChoices) {
+      this.emit('PickACard', messageSoFar, newChoices);
+    } else if (state.game.gameEndResults) {
```

Comment:
This line I had missed the .game part and it caused the game to get all the way to the end then fail to tell you if you won. It was terrible.